packages/theming-core/src/generics/themeRegistry.ts

```
@@ -0,0 +1,247 @@
+): IThemeRegistry<ITheme, IThemeDefinition> {
+  const graph: IThemeGraph<ITheme, IThemeDefinition> = {
+    _platform: { resolved: platformDefaults },
+    default: { parent: _platformKey }
+  };
+  return {
+    graph,
+    registerTheme: (definition: IThemeDefinition | ProcessTheme<ITheme, IThemeDefinition>, name?: string, parent?: string): void => {
+      _registerTheme<ITheme, IThemeDefinition>(graph, definition, name, parent);
+    },
+    getTheme: (name?: string): ITheme => {
+      return _getTheme<ITheme, IThemeDefinition>(graph, resolver, name);
+    },
+    updatePlatformDefaults: (newDefaults: ITheme): void => {
+      _updatePlatformDefaults<ITheme, IThemeDefinition>(graph, newDefaults);
+    }
+  };
+}
+
+/**
+ * Register a new theme by name.  This will invalidate the portion of the graph that is dependent on the new
+ * theme.  This will error on bad dependencies including cycles.
+ *
+ * @param graph - theme graph used for tracking dependencies
+ * @param definition - new theme definition or processing function to set in the theme
+ * @param name - name of the theme, if empty will be treated as 'default'
+ * @param parent - optional theme parent, if empty will be treated as 'default' if not default theme
+ */
+function _registerTheme<ITheme, IThemeDefinition>(
+  graph: IThemeGraph<ITheme, IThemeDefinition>,
+  definition: IThemeDefinition | ProcessTheme<ITheme, IThemeDefinition>,
+  name?: string,
+  parent?: string
+): void {
+  name = name || _defaultKey;
+  parent = (name === _defaultKey && _platformKey) || parent || _defaultKey;
+  if (_wouldCauseCycle(graph, name, parent)) {
+    throw new Error('Attempt to register a dependent theme that would cause a cycle');
+  } else if (!graph[parent]) {
+    throw new Error('Attempting to parent to an unknown theme');
+  } else if (name === _platformKey) {
+    throw new Error('A platform theme cannot be registered, call updatePlatformDefaults instead');
```

Would we really want to throw an exception for all of these cases? In general we seem to very rarely throw exceptions in Fabric, but I also haven't really seen a prescribed approach for throwing exceptions. In this case, I wonder if it would be better to log a warning but no-op for some of these rather than blow up the application.